### PR TITLE
feat: DatePicker setup basic single date picker

### PR DESCRIPTION
**Pull request by lloydstubber**
# Objective
This is the first piece of the DatePicker work. Which is creating a _very_ simple version of the component (bare bones API) and then the majority of the work being the styling of the input. The next piece will be styling the calendar and then getting into the meat of the API/documentation/tests side of things after that.

See entire epic for rest of work needing to be done: https://cultureamp.atlassian.net/jira/software/projects/KDS/boards/289?selectedIssue=KDS-62

# Motivation and Context
- Basic implementation of a single DatePicker with it's states styled to match the UI kit

# Notes
- Form primitive styling isn't the greatest. So we're just following suit here. We've tried to slim it down as much as possible and follow the TextField styling for hover/focus etc. as per @jmitsios 's instructions.
- Please disregard any API related work, this is just enough to get it rendering to style it. Although, the pattern of having it be a stateless component so we pass the date to it is how it'll work. So the story is very much how a consumer would use it.
- Label is closer in the implementation vs. the design but this matches how the TextField has it implemented. I think we'll need to update the label across the board rather than add padding to it per component. 

# Screenshots (if appropriate)
Default
image

Default on Hover
image

Default on Focus
image

Default w/ Value
image

Disbaled w/ Placeholder
image

Disabled w/ Value
image

### CODE REVIEW

File: packages/date-picker/src/DatePicker.scss [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  display: block;
+  font-size: 1rem;
+  line-height: 1.5;
+  padding: 0 $input-base-padding-horizontal;
+  height: $input-height;
+  width: 100%;
+  margin-bottom: 2px;
+
+  &.disabled {
+    &:not(.reversed) {
```
Comment (lloydstubber) [inline packages/date-picker/src/DatePicker.scss:44]: There is no reversed/secondary design at the minute but no doubt will be. Unsure if it's worth removing.

File: packages/date-picker/src/DatePicker.scss [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+.withStartIconAdornment {
+  .startIconAdornment {
+    @include vertically-center-icon;
+    @include ca-position($start: $ca-grid / 2);
+  }
+
+  .input {
+    @include ca-padding(
+      $start: 1rem + $input-icon-size,
```
Comment (lloydstubber) [inline packages/date-picker/src/DatePicker.scss:91]: 1rem is to handle the styling for icon at the start of input which isn't relative to `ca-grid`

**Review by ActuallyACat — CHANGES_REQUESTED** [review]

**Comment by jmitsios** [comment]
@lloydstubber

- _(See video attached)_ I'm noticing a situation where after I choose a date and I realise it wasn't the one I wanted.. I'd click back into the text field to re-trigger the calendar, but instead it just clears the focus state. Forcing another click to open it again.
- The calendar also seems to be overlapping the bottom of the focus ring too.
- I'm assuming to not make any design review comments about the calendar itself in the storybook link?
https://user-images.githubusercontent.com/95595231/151920540-075df21f-dba9-4325-b06a-33ecde218f0c.mov

**Review by jmitsios — APPROVED** [review]
The styling of the inputs is spot on 👍

File: packages/date-picker/src/DatePicker.scss [hunk 3]
```diff
@@ -0,0 +1,131 @@
+@import "~@kaizen/design-tokens/sass/border";
+@import "~@kaizen/design-tokens/sass/color";
+@import "~@kaizen/deprecated-component-library-helpers/styles/layout";
+@import "~@kaizen/deprecated-component-library-helpers/styles/color";
+@import "~@kaizen/component-library/styles/mixins/forms";
+@import "~@kaizen/draft-form/KaizenDraft/Form/Primitives/base";
+@import "~@kaizen/draft-button/KaizenDraft/Button/styles";
```
[the remark this answers is not in the document]
Comment (NatalieLoyzaga) [reply packages/date-picker/src/DatePicker.scss:7]: I have removed the button import as it was not being used

**Comment by lloydstubber** [comment]
Closed in favour of https://github.com/cultureamp/kaizen-design-system/pull/2511

**Pull request closed without merging**